pcdsdevices/fms.py

```diff
@@ -356,3 +359,73 @@ class LCP2(Device):
     air_temp_out_bot_f = Cpt(EpicsSignalRO, ':AirTempOutBot_CALC', kind='hinted')
     front_door_access = Cpt(EpicsSignalRO, ':FrontDoorAccess_CALC', kind='hinted')
     rear_door_access = Cpt(EpicsSignalRO, ':RearDoorAccess_CALC', kind='hinted')
+
+
+class SRCController(Device):
+    """
+    A Raritan SRC 800 controller. One unit required to connect
+    to all Raritan sensors.
+    """
+    src_model = Cpt(EpicsSignalRO, ':SRC800:MODEL_RBV', doc="Raritan 8 port Smart Sensor Controller")
+    serial_number = Cpt(EpicsSignalRO, ':SRC800:SERIAL_RBV', doc="Controller Serial Number")
+    number_sensors = Cpt(EpicsSignalRO, ':SRC800:MAX_SENSOR_NUM', doc="Number of sensors on controller")
+
+
+class RaritanSensor(Device):
+    """
+    Raritan Sensor meta data.
+    """
+    sensor_name = Cpt(EpicsSignalRO, ':NAME_RBV')
+    chain_position = Cpt(EpicsSignalRO, ':POSITION_RBV')
+    sensor_serial = Cpt(EpicsSignalRO, ':SERIAL_RBV')
+    state = Cpt(EpicsSignalRO, ':STATE_RBV')
+
+
+@reorder_components(
+    start_with=[
+        'sensor_name', 'leak_detection'
+    ]
+)
+class Floor(RaritanSensor):
+    """
+    Raritan Water/Leak Rope SmartSensor DX2-WSC-35-KIT
+    """
+    leak_detection = Cpt(EpicsSignalRO, ':SEVR_RBV', doc="Detects fluid leak, 0 dry, 2 wet")
```

Comment:
Does 1 mean error?

Reply:
```
/* Base is -1! */
static struct state {
    char *name;
    int   alarm;
} sensor_state[] = {
    { "unavailable",        NO_ALARM },
    { "open",               NO_ALARM },
    { "closed",             NO_ALARM },
    { "belowLowerCritical", MAJOR_ALARM },
    { "belowLowerWarning",  MINOR_ALARM },
    { "normal",             NO_ALARM },
    { "aboveUpperWarning",  MINOR_ALARM },
```
The value written to `SEVR` is the alarm state. I think 0 is `NO_ALARM` and 2 is `MAJOR_ALARM`. I've never seen a 1 for `MINOR_ALARM` which probably makes sense, since all leaks should be a `MAJOR_ALARM`. The alarm enums probably live here: `#include<alarm.h>` but I don't know where to find that file.